cmake/CPM.cmake

```diff
@@ -525,7 +525,12 @@ function(CPMAddPackage)
     list(APPEND CPM_ARGS_UNPARSED_ARGUMENTS SOURCE_DIR ${download_directory})
     if(EXISTS ${download_directory})
       # avoid FetchContent modules to improve performance
-      set(${CPM_ARGS_NAME}_BINARY_DIR ${CMAKE_BINARY_DIR}/_deps/${lower_case_name}-build)
+      if(DEFINED FETCHCONTENT_BASE_DIR)
+        # respect FETCHCONTENT_BASE_DIR if set
+        set(${CPM_ARGS_NAME}_BINARY_DIR ${FETCHCONTENT_BASE_DIR}/${lower_case_name}-build)
+      else()
+        set(${CPM_ARGS_NAME}_BINARY_DIR ${CMAKE_BINARY_DIR}/_deps/${lower_case_name}-build)
+      endif()
```

Comment:
How about merging the conditional here and below by defining and using a common variable `FETCH_CONTENT_ROOT` that is conditionally set to `${FETCHCONTENT_BASE_DIR}` or `${CMAKE_BINARY_DIR}/_deps` above? This would make it easier to use in other places and avoid future bugs.

Reply:
Done!